cli/azd/pkg/tools/docker/docker.go

```diff
@@ -90,11 +93,28 @@ func (d *docker) Build(
 
 	args = append(args, buildContext)
 
-	res, err := d.executeCommand(ctx, cwd, args...)
+	// Build and produce output
+	runArgs := exec.NewRunArgs("docker", args...).WithCwd(cwd)
+	d.console.StopSpinner(ctx, "", input.Step)
+	previewer := d.console.ShowPreviewer(ctx,
+		&input.ShowPreviewerOptions{
+			Prefix:       "    ",
+			MaxLineCount: 8,
+			Title:        "Docker Output",
+		})
+	runArgs.StdOut = previewer
+
+	_, err := d.commandRunner.Run(ctx, runArgs)
+	previewer.Stop(ctx)
```

Comment:
I don't think the concept of a previewer should exist directly in the tool.  Instead I would consider this tool taking an optional dependency on an `io.Writer` and can be injected via DI.

Then the preview experience can be controlled by an higher order component within the CLI.

Reply:
that would be cool. But a previwer-style can't be applied to any tool's output.
Only the tool knows what commands can support this style.

Reply:
I agree that only some commands make sense to display a preview / output of a tools command, but the implementation could be consistent.

Tools that want to participate in this simply just need to accept the `io.Writer` and then our CLI commands/components determine when to broadcast these messages to `stdout`.

Reply:
The previewer could always be listening, receiving the writes and just short circuit return when its not enabled.

Reply:
At that point we could even attach the `io.Writer` of the command previewer directly at the `CommandRunner` level and then just control when we display the output.

Reply:
It would be great to leverage this for hooks as well when the user opts to choose interactive mode.  Then we can have a nice and consistent experience and all we need to do is flip the flag since its already attached to the command runner.

Reply:
The previewer must be coordinated by the console, which is the Input/Output orchestrator.

If run a hook and expect to use the previewer, but there is a spinner running (we currently have this bug) there is an ugly output competing to show the output